File: test_search.py
import urllib.request, re, urllib.parse

query = "Zanjeer full movie"
url = "https://www.youtube.com/results?search_query=" + urllib.parse.quote(query)
headers = {
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36",
    "Accept-Language": "en-US,en;q=0.9"
}
req = urllib.request.Request(url, headers=headers)
with urllib.request.urlopen(req, timeout=10) as resp:
    html = resp.read().decode('utf-8', errors='ignore')

matches = re.findall(r'"videoId":"([a-zA-Z0-9_-]{11})".*?"title":\{"runs":\[\{"text":"(.*?)"\}\]', html)
print(f"Total raw matches: {len(matches)}")
seen = set()
for vid, title in matches:
    if vid not in seen:
        seen.add(vid)
        print(f"ID: {vid} | Title: {title.encode('ascii', 'replace').decode()}")

File: test_search.kt
import java.net.URL
import java.net.HttpURLConnection
import java.net.URLEncoder

fun searchYouTube(query: String) {
    val encoded = URLEncoder.encode("$query full movie", "UTF-8")
    val url = URL("https://www.youtube.com/results?search_query=$encoded")
    val conn = url.openConnection() as HttpURLConnection
    conn.setRequestProperty("User-Agent", "Mozilla/5.0 (Linux; Android 10; Mobile) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Mobile Safari/537.36")
    conn.connectTimeout = 10000
    conn.readTimeout = 10000
    val html = conn.inputStream.bufferedReader().use { it.readText() }
    
    val regex = Regex(""""videoId":"([a-zA-Z0-9_-]{11})".*?"title":\{"runs":\[\{"text":"(.*?)"\}\]\}""")
    val matches = regex.findAll(html).take(10).toList()
    println("Found ${matches.size} matches:")
    for (m in matches) {
        val vid = m.groupValues[1]
        val title = m.groupValues[2]
        println("ID: $vid | Title: $title")
    }
}
searchYouTube("Zanjeer")
